BridgeAppSDK/SBAUserWrapper+Bridge.swift

```diff
@@ -0,0 +1,311 @@
+            if let response = responseObject as? SBAUserSessionInfoWrapper {
+                self!.copyFromUserSession(response)
+            }
+            
+            if let consentSignature = self!.consentSignature where requiresConsent {
+                // If there is a consent signature object stored for this user then attempt
+                // sending consent once signed in.
+                self!.sendUserConsented(consentSignature, completion: completion)
+            }
+            else {
+                // otherwise, we are done. Set the flag that the consent has been verified and 
+                // call the completion
+                self!.consentVerified = !requiresConsent
+                self!.callCompletionOnMain(error, completion: completion)
+            }
+        }
+    }
+    
+    private func copyFromUserSession(response: SBAUserSessionInfoWrapper) {
+        
+        // Get the data groups from the response object
+        self.dataGroups = response.dataGroups
+        
+        // Get whether or not sharing is enabled and the sharing scope
+        self.dataSharingEnabled = response.dataSharingEnabled
+        if self.dataSharingEnabled {
+            self.dataSharingScope = response.dataSharingScope
+        }
+        
+        // Get the subpopulation consent status
+        self.subpopulationGuid = response.subpopulationGuid
+    }
+    
+    private func emailAndPasswordForExternalId(externalId: String) -> (String?, String?) {
+        
+        guard let emailFormat = SBAAppDelegate.sharedDelegate?.bridgeInfo.emailFormatForRegistrationViaExternalId else {
+            assertionFailure("'emailFormatForRegistrationViaExternalId' key missing from BridgeInfo")
+            return (nil, nil)
+        }
+        let passwordFormat = SBAAppDelegate.sharedDelegate?.bridgeInfo.passwordFormatForRegistrationViaExternalId ?? "%@"
+        
+        let email = NSString(format: emailFormat, externalId) as String
+        let password = NSString(format: passwordFormat, externalId) as String
+        
+        return (email, password)
+    }
+    
+    private func callCompletionOnMain(error: NSError?, completion: ((NSError?) -> Void)?) {
+        dispatch_async(dispatch_get_main_queue()) {
+            completion?(error)
+        }
+    }
+
+}
+
+protocol SBAUserSessionInfoWrapper : class {
+    var dataGroups : [String]? { get }
+    var dataSharingEnabled : Bool { get }
+    var dataSharingScope: SBBUserDataSharingScope { get }
+    var subpopulationGuid: String? { get }
+}
+
+extension NSDictionary: SBAUserSessionInfoWrapper {
+    
+    var dataGroups : [String]? {
+        return self["dataGroups"] as? [String]
+    }
+    
+    var dataSharingEnabled : Bool {
+        return self["dataSharing"] as? Bool ?? false
+    }
+    
+    var dataSharingScope: SBBUserDataSharingScope {
+        guard let sharingKey = self["dataGroups"] as? String where self.dataSharingEnabled else {
```

Comment:
I'm confused about the syntax here since I thought self["dataGroups"] should be an array of strings, not a single string... can you enlighten me as to what's going on here?